tests/permit-upsert.test.ts

```
@@ -0,0 +1,440 @@
+const mockSelect = jest.fn();
+const selectBuilder = {
+  eq: jest.fn(),
+  maybeSingle: mockMaybeSingle,
+};
+selectBuilder.eq.mockImplementation(() => selectBuilder);
+const updateResult: UpdateResponse = { data: [{ id: 123 }], error: null };
+const updateBuilder = {
+  eq: jest.fn(),
+  is: jest.fn(),
+  select: jest.fn(),
+  then: (resolve: (value: UpdateResponse) => void) => resolve(updateResult),
```

Biome (2.1.2)

[error] Do not add then to an object.



(lint/suspicious/noThenProperty)